src/components/shared/LanguageSelector/LanguageSelector.js

```diff
@@ -9,14 +9,16 @@ import { InjectAppServices } from '../../../services/pure-di';
  * @param { import('react-intl').InjectedIntl } props.intl
  */
 const LanguageSelector = ({
+  urlParameters,
   dependencies: {
     window: { Menubutton },
   },
 }) => {
   const intl = useIntl();
   const lang = intl.locale;
   const languageDropdown = useRef(null);
-
+  const langToChange = lang === 'en' ? 'es' : 'en';
+  urlParameters = urlParameters.replace(/(\?)?(&)?(lang=(en|es))?(&)?/g, '');
```

Comment:
I find this very strange, but I can't think of a better way to do it. 😅

Reply:
because we have many case to cover. so.. i delete the parameter first and then put again with all parameters